File: Length of Last Word/Length of Last Word.cpp
class Solution {
public:
    int lengthOfLastWord(const char *s) {
        if(!s || ((*s) == '\0'))  return 0;
        
        int len = 0;
        int prev = 0;
        while(s && ((*s) != '\0')){
            if((*s) != ' '){
                ++len;
                ++s;
            }
            else{
                prev = len;
                len = 0;
                while((*s) == ' ')
                    ++s;
            }
        }
        
        return (len) ? len : prev;
    }
};

File: Length of Last Word/Length of Last Word.java
public class Solution {
    public int lengthOfLastWord(String s) {
        int len = s.length();
        String word = new String();
        for(int i = 0; i < len; ){
            char c = s.charAt(i);
            if(c == ' '){
                do{
                    ++i;
                    if(i == len){
                        break;
                    }
                }while(s.charAt(i) == ' ');
                if(i == len){
                    return word.length();
                }
                else{
                    word = new String();
                }
            }
            else{
                word = word + c;
                ++i;
            }
        }
        return word.length();
    }
}
